src/app/ReadHandler.cpp

```diff
@@ -137,6 +159,14 @@ ReadHandler::~ReadHandler()
     InteractionModelEngine::GetInstance()->ReleaseAttributePathList(mpAttributePathList);
     InteractionModelEngine::GetInstance()->ReleaseEventPathList(mpEventPathList);
     InteractionModelEngine::GetInstance()->ReleaseDataVersionFilterList(mpDataVersionFilterList);
+
+// TODO (#27672): Enable when the ReportScheduler is implemented
+#if 0
+    if (nullptr != mObserver)
+    {
+        mObserver->OnReadHandlerRemoved(this);
```

Comment:
We can't put this in Close, because we can destroy a ReadHandler without calling Close.

The old timer cancellation guaranteed that the ReadHandler itself did not escape at that point, but the new callback is passing in the ReadHandler, so better to have it be at the beginning of the destructor.